internal/services/redis/redis_cache_resource.go

```diff
@@ -817,11 +817,17 @@ func expandRedisConfiguration(d *pluginsdk.ResourceData) (*redis.CommonPropertie
 	}
 
 	// RDB Backup
-	if v := raw["rdb_backup_enabled"].(bool); v {
-		if connStr := raw["rdb_storage_connection_string"].(string); connStr == "" {
-			return nil, fmt.Errorf("The rdb_storage_connection_string property must be set when rdb_backup_enabled is true")
+	// the "output.RdbBackupEnabled" should be set value when the rdb_backup_enabled is set to false in config since not setting false means it could not be disabled using patch to update the Redis Cache.
+	// d.GetOk cannot identify if user sets the property that is bool type and `rdb_backup_enabled` is set as `false`. So it has to identify it using `d.GetRawConfig()`
+	if v := d.GetRawConfig().AsValueMap()["redis_configuration"].AsValueSlice(); len(v) > 0 && !v[0].AsValueMap()["rdb_backup_enabled"].IsNull() {
```

Comment:
rather than using `GetRawConfig` - we should update this to use `d.GetOkExists` - since this approach can be nil/crash at multiple points along this expression

Reply:
@tombuildsstuff thanks for your time to review this PR. If we use `if v, ok := d.GetOkExists("redis_configuration.0.rdb_backup_enabled"); ok{`  instead of `GetRawConfig`, the vaule of ok is true when the `rdb_backup_enabled` is removed in tf config. But the `GetRawConfig` is null (meaning false) in this case. 

Using `GetOkExists` means that if user specifies `rdb_backup_enabled` as true in creating a resource, and then removes `rdb_backup_enabled` in tf config to update the resource, the  `rdb_backup_enabled`  would be set to false.  In fact, the behavior of API in this case, the  `rdb_backup_enabled`  should remain true. That's why I use   `GetRawConfig` ranther than `GetOkExists`.  BTW, if rdb_backup_enabled is not specified, API return nil. 

So I assume that the `GetRawConfig` could not be replaced by `GetOkExists`. Please feel free let me know your feedback on this case.

Reply:
@sinbai if you are going to do this you must be checking each conversion for a nil/crash. right now if anything changes/something unexpected happens this code will panic